src/python/monsoon_wang/scripts/mpindex_compute.py

```diff
@@ -152,71 +111,19 @@
 except BaseException:
     pass
 
-modpathall = modpath.replace('MODS', '*')
-lst = glob.glob(modpathall)
-# CONFIRM DATA FOR MODS IS AVAIL AND REMOVE THOSE IT IS NOT
-
-gmods = []  # "Got" these MODS
-print("MODS:", mods)
-print("LST:", lst)
-for mod in mods:
-    for l in lst:
-        l1 = modpath.replace('MODS', mod)
-        print("L!:", l1)
-        if os.path.isfile(l1) is True:
-            if mod not in gmods:
-                gmods.append(mod)
-
-if args.experiment == 'historical' and mods is None:
-    gmods = [
-        'ACCESS1-0',
-        'ACCESS1-3',
-        'bcc-csm1-1',
-        'bcc-csm1-1-m',
-        'BNU-ESM',
-        'CanCM4',
-        'CanESM2',
-        'CCSM4',
-        'CESM1-BGC',
-        'CESM1-CAM5',
-        'CESM1-FASTCHEM',
-        'CESM1-WACCM',
-        'CMCC-CESM',
-        'CMCC-CM',
-        'CMCC-CMS',
-        'CNRM-CM5-2',
-        'CNRM-CM5',
-        'CSIRO-Mk3-6-0',
-        'FGOALS-g2',
-        'FIO-ESM',
-        'GFDL-CM2p1',
-        'GFDL-CM3',
-        'GFDL-ESM2G',
-        'GFDL-ESM2M',
-        'GISS-E2-H',
-        'GISS-E2-H-CC',
-        'GISS-E2-R',
-        'GISS-E2-R-CC',
-        'HadCM3',
-        'HadGEM2-AO',
-        'HadGEM2-CC',
-        'HadGEM2-ES',
-        'inmcm4',
-        'IPSL-CM5A-LR',
-        'IPSL-CM5A-MR',
-        'IPSL-CM5B-LR',
-        'MIROC4h',
-        'MIROC5',
-        'MIROC-ESM',
-        'MIROC-ESM-CHEM',
-        'MPI-ESM-LR',
-        'MPI-ESM-MR',
-        'MPI-ESM-P',
-        'MRI-CGCM3',
-        'MRI-ESM1',
-        'NorESM1-M',
-        'NorESM1-ME']
+gmods = [] # TRAP LIST OF MODELS AVAILABLE WHICH MAY BE LESS THAN REQUESTED FROM PARAM FILE.
+for m in mods:
+  modpath.model = m
+  if os.path.isfile(modpath()) is True and m not in gmods:  
```

Comment:
probably could make gmods a `set()` here would avoid checking if m already in it.